consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/statemachine/RoundState.java

```diff
@@ -79,13 +82,15 @@ public boolean setProposedBlock(final SignedData<ProposalPayload> msg) {
   public void addPrepareMessage(final SignedData<PreparePayload> msg) {
     if (!proposalMessage.isPresent() || validator.validatePrepareMessage(msg)) {
```

Comment:
re-reading this conditional kinda hurt my head, its correct, but not obviously.
May be a symptom of the design rather than the code ... but ... :/
Rather see it say "If(Proposal.isPresent() && isValid) store; else if(!Proposal) store.